### PR TITLE
Add integration test for VolumeRestriction in requeueing scenarios

### PR DESCRIPTION
**Pull request by utam0k**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Add integration test for VolumeRestriction in requeueing scenarios

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes/kubernetes/issues/122305

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/integration/scheduler/queue_test.go [hunk 1]
```diff
@@ -1231,6 +1231,71 @@ func TestCoreResourceEnqueue(t *testing.T) {
...
+				if _, err := testCtx.ClientSet.CoreV1().PersistentVolumeClaims(testCtx.NS.Name).Create(testCtx.Ctx, pvc2, metav1.CreateOptions{}); err != nil {
+					return fmt.Errorf("failed to add pvc1: %w", err)
+				}
+				return nil
+			},
+			wantRequeuedPods:          sets.New("pod1"),
+			enableSchedulingQueueHint: []bool{true},
+		},
+		{
+			name:         "Pod rejected by the VolumeRestriction plugin is requeued when the the PVC bound to the pod is added",
```
Comment (AxeZhan) [inline test/integration/scheduler/queue_test.go:1265]: case name not changed.

**Comment by AxeZhan** [comment]
Maybe we can also add cases for `not requeued` scenario?

File: test/integration/scheduler/queue_test.go (hunk 1 above)
Comment (utam0k) [reply test/integration/scheduler/queue_test.go]: 🤦

**Comment by utam0k** [comment]
> Maybe we can also add cases for `not requeued` scenario?

Added the simple scenario for `not requeued`.

**Comment by AxeZhan** [comment]
/lgtm 
For `not requeued` case, we may need a follow-up in #127607

/cc @sanposhiho

File: test/integration/scheduler/queue_test.go [hunk 2]
```diff
@@ -1231,6 +1231,106 @@ func TestCoreResourceEnqueue(t *testing.T) {
...
+				if err := testCtx.ClientSet.CoreV1().Pods(testCtx.NS.Name).Delete(testCtx.Ctx, "pod1", metav1.DeleteOptions{GracePeriodSeconds: new(int64)}); err != nil {
+					return fmt.Errorf("failed to delete pod1: %w", err)
+				}
+				return nil
+			},
+			wantRequeuedPods:          sets.New("pod2"),
+			enableSchedulingQueueHint: []bool{true},
+		},
+		{
+			name:         "Pod rejected by the VolumeRestriction plugin isn't requeued when the deleted pod has the different PVC",
```
Comment (sanposhiho) [inline test/integration/scheduler/queue_test.go:1300]: can we simply merge this test case into the above one?

File: test/integration/scheduler/queue_test.go [hunk 3]
```diff
@@ -1231,6 +1231,106 @@ func TestCoreResourceEnqueue(t *testing.T) {
...
+			initialPVs: []*v1.PersistentVolume{
+				st.MakePersistentVolume().
+					Name("pv1").
+					AccessModes([]v1.PersistentVolumeAccessMode{v1.ReadWriteOnce}).
+					Capacity(v1.ResourceList{v1.ResourceStorage: resource.MustParse("1Mi")}).
+					HostPathVolumeSource(&v1.HostPathVolumeSource{Path: "/tmp", Type: ptr.To(v1.HostPathDirectoryOrCreate)}).
+					Obj(),
+			},
+			pods: []*v1.Pod{
+				st.MakePod().Name("pod1").Container("image").PVC("pvc1").Obj(),
```
Comment (sanposhiho) [inline test/integration/scheduler/queue_test.go:1246]: add a pod which wouldn't be requeued with the pvc addition.

File: test/integration/scheduler/queue_test.go (hunk 2 above)
Comment (utam0k) [reply test/integration/scheduler/queue_test.go:1300]: Make sense.

**Comment by sanposhiho** [comment]
LGTM, but please rebase; https://github.com/kubernetes/kubernetes/pull/127607 is now merged and `triggeredFn` got a small change.

**Comment by utam0k** [comment]
> LGTM, but please rebase; #127607 is now merged and `triggerFn` got a small change.

Sure! Done.

**Review by sanposhiho — APPROVED** [review]
/lgtm
/approve

**Comment by utam0k** [comment]
/sig scheduling 

The reality is that sig-scheduling members are reviewing it.

**Pull request merged**